wavesexchange_topic/src/lib.rs

```diff
@@ -303,10 +316,64 @@ mod parse_and_format {
                             return Err(TopicParseError::InvalidLeasingBalanceTopic);
                         }
                     }
+                    TopicKind::Pairs => {
+                        Topic::extract_exchange_pairs_from_query(&url)?;
+                    }
                 }
 
                 Ok(())
             }
+
+            pub fn extract_exchange_pairs_from_query<'a>(
+                url: &'a Url,
+            ) -> Result<Vec<ExchangePairs>, TopicParseError> {
+                let mut ret = vec![];
+
+                match url.path_segments() {
+                    Some(mut parts) => {
+                        // topic://pairs/<amount_asset_id>/<price_asset_id>
+                        let amount_asset = parts.next();
+                        let price_asset = parts.next();
+
+                        if amount_asset.is_some() && price_asset.is_some() {
+                            ret.push({
```

Comment:
Maybe worth checking `!parts.has_next()` and `url.query().is_empty()` and so forth (see above examples of parsing different kinds of topic URLs) so that URL is verified correct.